tgui/packages/common/storage.js

```diff
@@ -150,8 +150,47 @@ class IndexedDbBackend {
   }
 }
 
-export const storage = (
-  testIndexedDb() && new IndexedDbBackend()
-  || testLocalStorage() && new LocalStorageBackend()
-  || new MemoryBackend()
-);
+/**
+ * Web Storage Proxy object, which selects the best backend available
+ * depending on the environment.
+ */
+class StorageProxy {
+  constructor() {
+    this.backendPromise = (async () => {
+      if (testIndexedDb()) {
+        try {
+          const backend = new IndexedDbBackend();
+          await backend.dbPromise;
+          return backend;
+        }
+        catch {}
```

Comment:
Silent catch bad.

Reply:
upstream issue?

Reply:
Seems so.

Reply:
@stylemistake Is this something you'd want us to fix upstream?

Reply:
upstream reasoning: below lines and `It falls back gracefully to LocalStorageBackend, and MemoryBackend if browser throws an error in the process.`